src/arrayMethodJoin.js

```diff
@@ -4,9 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let resultJoin = '';
+
+    for (let i = 0; i <= this.length - 1; i++) {
```

Comment:
```suggestion
    for (let i = 0; i < this.length; i++) {
```
It's not a mistake, but usually arrays are looped to values < length.